src/main/java/Calculations/GPACalculation.java

```diff
@@ -0,0 +1,167 @@
+            return "A";
+        }else if(percentage1 >= 80){
+            return "A-";
+        }else if(percentage1 >= 77){
+            return "B+";
+        }else if(percentage1 >= 73){
+            return "B";
+        }else if(percentage1 >= 70){
+            return "B-";
+        }else if(percentage1 >= 67){
+            return "C+";
+        }else if(percentage1 >= 63){
+            return "C";
+        }else if(percentage1 >= 60){
+            return "C-";
+        }else if(percentage1 >= 57){
+            return "D+";
+        }else if(percentage1 >= 55){
+            return "D";
+        }else if(percentage1 >= 50){
+            return "D-";
+        }else {
+            return "F";
+        }
+    }
+
+    /**
+     * Takes list of double consisting of course marks in terms of percentage and converts to list of lettered Grades
+     * using helper function percentToGrade
+     * @param marksPercentage2 a list of double
+     * @return gradelist which is a list of string consisting of lettered Grade
+     */
+    public static List<String> computeGrade(List<Double> marksPercentage2) {
+        List<String> gradelist = new ArrayList<>();
+        for(int i = 0; i < marksPercentage2.size(); i++) {
+            gradelist.add(percentToGrade(marksPercentage2.get(i)));
+        }
+        //System.out.println(gradelist);
+        return gradelist;
+    }
+
+    /**
+     * Takes a list of double of markpercentage (course marks) and a list of courseCredit (0.5 or 1 depending on
+     * half-year or full-year). Computes overall GPA accounting for course credit. GPA is calculated by first
+     * taking sum of credits, then converting markpercentage into gpalist, then creating a new list consisting of
+     * gpa x course credit, and taking sum of this and dividing by sum of credits.
+     *
+     * @param markpercentage is a list of double consisting of course marks
+     * @param courseCredits is a list of double consisting of course credits
+     * @return a double of the overall GPA to 2 decimal places
+     */
+    public static double overallGPA(List<Double> markpercentage, List<Double> courseCredits) {
+        double sumGPA = 0;
+        double sumCredits = 0;
+        double totalGpa;
+        List<Double> gpalist = new ArrayList<>();
+        List<Double> overallGrade = new ArrayList<>();
+
+        for(int i = 0; i < courseCredits.size(); i++) {
+            sumCredits += courseCredits.get(i);
+        }
+        for(int j = 0; j < markpercentage.size(); j++) {
+            gpalist.add(computationGPA(markpercentage.get(j)));
+        }
+        for(int s = 0; s < gpalist.size(); s ++){
+            overallGrade.add(gpalist.get(s) * courseCredits.get(s));
+        }
+        for(int v = 0; v < overallGrade.size(); v++) {
+            sumGPA += overallGrade.get(v);
+        }
+        totalGpa = Math.round((sumGPA / sumCredits) * 100.0) / 100.0;
+        //System.out.println(totalGpa);
+        return totalGpa;
```

Comment:
I don't believe that this function is doing its computations correctly. Again, consider writing a test suite to ensure everything is working properly.